File: 3442_Maximum_Difference_Between_Even_and_Odd_Frequency_I.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int maxDifference(String s) {
        Map<Character, Integer> charFrequencies = new HashMap<>();
        for (char c : s.toCharArray()) {
            charFrequencies.put(c, charFrequencies.getOrDefault(c, 0) + 1);
        }

        int maxOddFrequency = 0;
        int minEvenFrequency = Integer.MAX_VALUE; 

        for (int frequency : charFrequencies.values()) {
            if (frequency % 2 != 0) { // Odd frequency
                maxOddFrequency = Math.max(maxOddFrequency, frequency);
            } else { // Even frequency
                minEvenFrequency = Math.min(minEvenFrequency, frequency);
            }
        }
        return maxOddFrequency - minEvenFrequency;
    }
}

File: 3442_Maximum_Difference_Between_Even_and_Odd_Frequency_I.cpp
class Solution {
public:
    int maxDifference(string s) {
        std::vector<int> frq(26, 0);
        for (char ch : s) {
            frq[ch - 'a']++;
        }

        int max_odd_freq = 0; // maximum odd frequency
        int min_even_freq = 101; // minimum even frequency (s.length is max 100)

        for (int f : frq) {
            if (f == 0) continue; 
            if (f % 2 != 0) {
                max_odd_freq = std::max(max_odd_freq, f);
            } else {
                min_even_freq = std::min(min_even_freq, f);
            }
        }
        return max_odd_freq - min_even_freq;
    }
};
